test/issue8/proto.pb.go

```diff
@@ -64,6 +66,11 @@ func (m *Foo) Unmarshal(data []byte) error {
 		}
 		fieldNum := int32(wire >> 3)
 		wireType := int(wire & 0x7)
+
+		if fn := int(fieldNum - 1); fn <= 0 {
```

Comment:
@anton-povarov could you help us fix that, please? We copied the `Unmarshal` changes from [your suggestion](https://github.com/gogo/protobuf/compare/master...anton-povarov:unmarshaller_required_fields), as found in #26. I don't think I completely understand what's going on here. It looks like that line is being written by [this line](https://github.com/gogo/protobuf/compare/master...anton-povarov:unmarshaller_required_fields#diff-e0188cb7643b9790acd32abc395a5659R719).

Thank you!